source/cppfront.cpp

```diff
@@ -84,7 +84,12 @@ auto main(
         }
 
         //  Load + lex + parse + sema
-        cppfront c(arg.text);
+        cppfront c = [&]() -> cppfront {
```

Comment:
Would it be less change to handle this split in the `load()` function instead? https://github.com/hsutter/cppfront/blob/main/source/io.h#L885 

It would need a little change where it currently checks for the file extension, but the rest may just work.

Reply:
Ah yea you're right. I was struggling a bit to figure out where to split this but I think that would make a lot more sense. Ill go ahead and give that a try. Thanks!

Reply:
Thanks for that suggestion, that greatly simplified the code!

Reply:
You're welcome.